File: map_python/src/Container/operation.py
from dataclasses import dataclass
from typing import Callable, Generic, TypeVar

T = TypeVar("T")
U = TypeVar("U")


@dataclass(slots=True)
class Operation(Generic[T, U]):
    name: str
    instruction: Callable[[T], U]


@dataclass(slots=True)
class ExpandOperation(Generic[T, U]):
    name: str
    instruction: Callable[[T], list[U]]

File: map_ts/src/Container/operation.ts
// operation.ts

export type Operation<TInput, TOutput = TInput> = {
  name: string;
  instruction: (input: TInput) => TOutput;
};

export type ExpandOperation<TInput, TOutput = TInput> = {
  name: string;
  instruction: (input: TInput) => TOutput[];
};